packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -53,15 +53,19 @@ export const vModelText: ModelDirective<
 > = {
   created(el, { modifiers: { lazy, trim, number } }, vnode) {
     el[assignKey] = getModelAssigner(vnode)
-    const castToNumber =
-      number || (vnode.props && vnode.props.type === 'number')
+    const vnodeType = vnode.props && vnode.props.type
+    const castToNumber = number || vnodeType === 'number'
+    const castToTimeStamp =
+      number && (vnodeType === 'date' || vnodeType === 'datetime-local')
     addEventListener(el, lazy ? 'change' : 'input', e => {
       if ((e.target as any).composing) return
       let domValue: string | number = el.value
       if (trim) {
         domValue = domValue.trim()
       }
-      if (castToNumber) {
+      if (castToTimeStamp) {
+        domValue = new Date(domValue).getTime()
```

Comment:
This line will cause an error on iOS devices.
should be `new Date('2024-03-15 16:49:20'.replace(/-/g,'\/'))`